app/models/order_detail.rb

```diff
@@ -843,10 +844,17 @@ def manual_fulfilled_at=(string)
     @manual_fulfilled_at = ValidFulfilledAtDate.parse(string)
   end
 
+  def time_data
+    @time_data ||= product.time_data_for(self)
+  end
+
   private
 
   def has_completed_reservation?
-    !product.is_a?(Instrument) || (reservation && (reservation.canceled_at || reservation.actual_end_at || reservation.reserve_end_at < Time.zone.now))
+    return true unless product.timed?
```

Comment:
This was a first attempt at using `time_data` in actual business logic. This will likely change in the future (one thing I'm considering is a NullObject for item/service time_data).